rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/NSCreationServlet.java

```diff
@@ -150,28 +150,20 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
+                                                                                       nsName,
+                                                                                       infra,
+                                                                                       toArray(infraParams),
+                                                                                       toArray(infraFileParams),
+                                                                                       policy,
+                                                                                       toArray(policyParams),
+                                                                                       toArray(policyFileParams),
+                                                                                       nodesRecoverable);
+
             if (deployNodeSource) {
-                jsonResult = ((RMServiceImpl) RMServiceImpl.get()).createNodeSource(sessionId,
-                                                                                    nsName,
-                                                                                    infra,
-                                                                                    toArray(infraParams),
-                                                                                    toArray(infraFileParams),
-                                                                                    policy,
-                                                                                    toArray(policyParams),
-                                                                                    toArray(policyFileParams),
-                                                                                    nodesRecoverable);
-            } else {
-                jsonResult = ((RMServiceImpl) RMServiceImpl.get()).defineNodeSource(sessionId,
-                                                                                    nsName,
-                                                                                    infra,
-                                                                                    toArray(infraParams),
-                                                                                    toArray(infraFileParams),
-                                                                                    policy,
-                                                                                    toArray(policyParams),
-                                                                                    toArray(policyFileParams),
-                                                                                    nodesRecoverable);
+                jsonResult = ((RMServiceImpl) RMServiceImpl.get()).deployNodeSource(sessionId, nsName);
             }
+
             JSONObject json = new JSONObject(jsonResult);
             if (json != null) {
                 if (json.has("result")) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "result" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)